### PR TITLE
Cache for textfragments #433

### PR DESCRIPTION
**Pull request by Mathriok**
Lines read from files should now be saved in a chunk in cache when opened. When scrolling, lines should get sent from cache if they exist, otherwise another chunk will be read from file and stored in cache. See if this works on your machine!

### CODE REVIEW

File: src/js/adapters/fileReader.test.js [hunk 1]
```diff
@@ -115,14 +71,98 @@ describe('countLinesInBuffer', () => {
...
+  //     linesEndAt: 463
+  //   };
+  //   const testResult = await readDataFromByte(
+  //     filePath,
+  //     startReadFromByte,
+  //     numberOfBytes
+  //   );
+  //   expect(testResult.startByteOfLines[0]).not.toEqual(startReadFromByte);
+  //   expect(testResult).toEqual(expectedResult);
+  // });
```
Comment (MelindaSW) [inline src/js/adapters/fileReader.test.js]: These tests could be uncommented. As well as the last one. Tried running them and they went through for me.

File: src/js/view/components/LogViewer.js [hunk 2]
```diff
@@ -257,6 +259,8 @@ const LogViewer = props => {
           Math.round(logSize - scrollPosition),
           props.nrOfLinesInViewer
         );
+        let logSizeMinusScrollPos = Math.round(logSize - scrollPosition);
+        console.log({ logSizeMinusScrollPos });
```
Comment (MelindaSW) [inline src/js/view/components/LogViewer.js]: Found a couple of console.logs that could probably be removed, here and on row 17.

File: src/js/engine/cache.js [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+const flushCacheForOneFile = filepath => {
+  delete cache[filepath];
+};
+
+const cacheInit = cache;
+const checkIfCacheIsWithinSizeLimit = (cache = cacheInit) => {
+  /*check if cache has reached the limit of 100mb (a hundred millon bytes)*/
+  const cacheSize = Buffer.byteLength(JSON.stringify(cache), 'utf8');
+  const sizeLimit = 100000000;
+  // console.log({ withinLimit: cacheSize < sizeLimit });
```
Comment (MelindaSW) [inline src/js/engine/cache.js]: this could be deleted as well

File: src/js/adapters/fileReader.test.js (hunk 1 above)
Comment (Mathriok) [reply src/js/adapters/fileReader.test.js]: Aha! I uncommented them and tried running, but they failed at a couple of places. I did some minor changes in them in the latest commit and they work now for me. Could you see if they still work for you?

**Comment by MelindaSW** [comment]
I have scrolled back and forth and tried it with different files, and with running files. Have not seen anything wrong yet. Would like to have someone else than me reviewing this as well before approving though.

**Pull request merged**